File: apps/embed/src/xrpc/getProfile.ts
import { ROCKSKY_API_URL } from "../consts";
import type { Profile } from "../types/profile";

export default async function getProfile(handle: string) {
  const res = await fetch(
    `${ROCKSKY_API_URL}/xrpc/app.rocksky.actor.getProfile?did=${handle}`,
  );

  const profile = (await res.json()) as Profile;

  return { profile, ok: res.ok };
}

File: sdk/elixir/examples/get_profile.exs
# Fetch a public Rocksky profile.
#
#   ROCKSKY_DID=alice.bsky.social mix run examples/get_profile.exs

did = System.get_env("ROCKSKY_DID") || raise "set ROCKSKY_DID"

client =
  Rocksky.new(
    base_url: System.get_env("ROCKSKY_BASE_URL", "https://api.rocksky.app"),
    token: System.get_env("ROCKSKY_TOKEN")
  )

case Rocksky.Actor.get_profile(client, did: did) do
  {:ok, profile} ->
    IO.puts("Profile for #{did}:")
    IO.inspect(profile, pretty: true, limit: :infinity)

  {:error, err} ->
    IO.warn("failed: " <> Exception.message(err))
    System.halt(1)
end
